sdk/nodejs/runtime/stack.ts

```diff
@@ -25,14 +25,21 @@ import { getRootResource, isQueryMode, setRootResource } from "./settings";
  */
 export const rootPulumiStackTypeName = "pulumi:pulumi:Stack";
 
+let stackResource: Stack | undefined;
+
+// Get the root stack resource for the current stack deployment
+export function getStackResource(): Stack | undefined {
```

Comment:
actually, it feels like we shoudl always be able to get this.  should we throw in this method if this is called in a scenario where stackResource was not set?  That would indicate *this* runInpulumiStack was not called, which would be weird if we were then running the same code in resource-construction couldn't access this.

Reply:
Done.  I'm slightly nervous about this - but okay with being strict here and seeing if we encounter places this cause problems.